editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -152,17 +156,29 @@ private void configureFacet(MavenProject mavenProject, IProject project, IProgre
 	}
 
 	private void installDefaultFacets(IProject project, MavenProject mavenProject, IFacetedProject fproj, IProgressMonitor monitor)
-			throws CoreException {
-		String camelVersion = getCamelVersion(mavenProject);
-		if (camelVersion != null) {
-			IFacetedProjectWorkingCopy fpwc = fproj.createWorkingCopy();
-			installFacet(fproj, fpwc, javaFacet, javaFacet.getLatestVersion(), monitor);
-			installFacet(fproj, fpwc, m2eFacet, m2eFacet.getLatestVersion(), monitor);
-			installFacet(fproj, fpwc, utilFacet, utilFacet.getLatestVersion(), monitor);
-			installCamelFacet(fproj, fpwc, camelVersion, monitor);
-			fpwc.commitChanges(monitor);
-			configureNature(project, mavenProject, monitor); 
-		}
+	        throws CoreException {
+	    String camelVersion = getCamelVersion(mavenProject);
+	    if (camelVersion != null) {
+	        IFacetedProjectWorkingCopy fpwc = fproj.createWorkingCopy();
+
+	        // adjust facets we install based on the packaging type we find
+	        String packaging = null;
+	        if (mavenProject.getPackaging() != null) {
```

Comment:
any reason why the packaging is calculated here? and not just before we need it?

If no, what do you think of grouping them and perhaps even extracting a method for configureFacetDependingOnFacet?